src/Symfony/Component/Form/AbstractRendererEngine.php

```diff
@@ -171,7 +171,7 @@ private function loadResourceForBlockNameHierarchy($cacheKey, FormView $view, ar
             // The next two if statements contain slightly duplicated code. This is by intention
             // and tries to avoid execution of unnecessary checks in order to increase performance.
 
-            if (isset($this->resources[$cacheKey][$parentBlockName])) {
+            if (isset($this->resources[$cacheKey][$parentBlockName]) && $this->resources[$cacheKey][$parentBlockName] !== false) {
```

Comment:
This change is missing a corresponding test.

Also you are reducing the impact of a performance optimization here. Could you benchmark the result for big forms?